images/kube-webhook-certgen/rootfs/cmd/patch.go

```diff
@@ -9,49 +13,109 @@ import (
 
 var patch = &cobra.Command{
 	Use:    "patch",
-	Short:  "Patch a validatingwebhookconfiguration and mutatingwebhookconfiguration 'webhook-name' by using the ca from 'secret-name' in 'namespace'",
-	Long:   "Patch a validatingwebhookconfiguration and mutatingwebhookconfiguration 'webhook-name' by using the ca from 'secret-name' in 'namespace'",
-	PreRun: prePatchCommand,
+	Short:  "Patch a ValidatingWebhookConfiguration, MutatingWebhookConfiguration or APIService 'object-name' by using the ca from 'secret-name' in 'namespace'",
+	Long:   "Patch a ValidatingWebhookConfiguration, MutatingWebhookConfiguration or APIService 'object-name' by using the ca from 'secret-name' in 'namespace'",
+	PreRun: configureLogging,
 	Run:    patchCommand,
 }
 
-func prePatchCommand(cmd *cobra.Command, args []string) {
-	configureLogging(cmd, args)
-	if !cfg.patchMutating && !cfg.patchValidating {
-		log.Fatal("patch-validating=false, patch-mutating=false. You must patch at least one kind of webhook, otherwise this command is a no-op")
+type PatchConfig struct {
+	PatchMutating      bool
+	PatchValidating    bool
+	PatchFailurePolicy string
+	APIServiceName     string
+	WebhookName        string
+
+	SecretName string
+	Namespace  string
+
+	Patcher Patcher
+}
+
+type Patcher interface {
```

Comment:
Do we intend to extend this program to patch not only k8s stuff? Otherwise I would vote into keeping this simple, saying that in PatchConfig we have a `type k8s struct` and be happy :D

Reply:
This is to avoid directly lower level dependency + to allow testing existing logic which is in cmd package. Without this, it would be much more difficult.